lib/json/JsonBonus.cpp

```diff
@@ -886,4 +886,19 @@ CSelector JsonUtils::parseSelector(const JsonNode & ability)
 	return ret;
 }
 
+std::shared_ptr<Bonus> JsonUtils::parseChargeBasedSpell(const JsonNode & ability)
+{
+	auto spellBonus = std::make_shared<Bonus>();
+	spellBonus->type = BonusType::SPELL;
+	LIBRARY->identifiers()->requestIdentifier( "spell", ability["spell"], [&spellBonus](int32_t identifier)
```

Comment:
Every time I try to parse a spell I get "-1", could you take a look at what's wrong @IvanSavenko ?
I'm trying to parse this structure:
```
		"instanceBonuses" : {
			"chargeBasedSpell" : {
					"spell" : "spell.bless",
					"chargeCost" : 2,
					"level" : 3
			}
		},
```

Reply:
I'm using a custom object "chargeBasedSpell" here, see config/schemas/artifact.json. In ability["spell"] there is a line exactly the same as in a regular bonus. But parsing works there with common binus spell.

Reply:
Ok. But after loading is completed, I still don't get the spell. And I don't see any difference between parsing in JsonUtils::parseBonus -> loadBonusSubtype and here. But it works there, and not here.

Reply:
The question is closed

Reply:
> `requestIdentifier` is async method

Thanks for the tip